flow/project.py

```diff
@@ -236,6 +251,119 @@ def make_bundles(operations, size=None):
     return _make_bundles(operations, size)
 
 
+class aggregate(object):
+    """Decorator for operation functions that are aggregate operations.
+
+    .. code-block:: python
+        example_aggregate = aggregate()
+
+        @example_aggregate
+        @FlowProject.operation
+        def foo(jobs):
+            return len(jobs)
+
+    :param groupper:
+        Information on how to aggregate jobs.
+    :type groupper:
+        callable
+    :param sort:
+        Sort jobs by a statepoint parameter.
+    :type sort:
+        str
+    :param reverse:
+        States whether sorting is to be done in ascending or descending order.
+    :type reverse:
+        bool
+    """
+
+    def __init__(self, grouper=None, sort=None, reverse=False):
+        if grouper is None:
+            def grouper(jobs):
+                yield jobs
+
+        def key_sort(job):
+            try:
+                return job.sp[sort]
+            except KeyError:
+                raise KeyError("The key '{}' was not found in statepoint "
+                               "parameters of the job {}.".format(sort, job))
+        self.grouper = grouper
+        self.sort = None if sort is None else functools.partial(sorted,
```

Comment:
Likewise, here. In general, if the user shouldn't change it, it should be underscored.

Reply:
Okay!